evalml/tests/component_tests/test_undersampler.py

```diff
@@ -74,3 +75,55 @@ def test_undersample_imbalanced(data_type, make_data_type):
 
     np.testing.assert_equal(X, transform_X.to_dataframe().values)
     np.testing.assert_equal(y, transform_y.to_series().values)
+
+
+@pytest.mark.parametrize("dictionary,msg", [({'majority': 0.5}, "Lengths are diff"),
+                                            ({'minority': 1}, "Lengths are diff"),
+                                            ({0: 1, 1: 0.1}, "Dictionary keys are different from"),
+                                            ({1: 0.1}, "Lengths are diff")])
+def test_undersampler_sampling_dict_errors(dictionary, msg):
+    X = np.array([[i] for i in range(1000)])
+    y = np.array(["minority"] * 150 + ["majority"] * 850)
+
+    undersampler = Undersampler(sampling_ratio_dict=dictionary)
+    with pytest.raises(ValueError, match=msg):
+        undersampler.fit_transform(X, y)
+
+
+@pytest.mark.parametrize("dictionary,dictionary_values", [({0: 1, 1: 0.5}, {0: 150, 1: 300}),
+                                                          ({0: 1, 1: 0.25}, {0: 150, 1: 600}),
+                                                          ({0: 1, 1: 0.1}, {0: 150, 1: 850}),
+                                                          ({0: 0.1, 1: 0.1}, {0: 150, 1: 850}),
+                                                          ({0: 0.1, 1: 1}, {0: 150, 1: 150})])
```

Comment:
Can you explain these tests to me?

Reply:
Yep! The first dictionary is the ratio dict that a user will pass in, while the second dictionary is the resulting dictionary we want to post-conversion. The first one holds the ratios, while the second one has the number of samples that we want in the final result.

Renamed the dictionaries for clarity!